tests/system/aiplatform/e2e_base.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import abc
+import importlib
+import os
+import pytest
+import uuid
+
+from google.cloud import aiplatform
+from google.cloud import storage
+from google.cloud.aiplatform import initializer
+
+_PROJECT = os.getenv("BUILD_SPECIFIC_GCLOUD_PROJECT")
+_LOCATION = "us-central1"
+
+
+class TestEndToEnd:
+    @property
+    @abc.abstractmethod
+    def _temp_prefix(cls) -> str:
+        """Prefix to staging bucket and display names created by this end-to-end test.
+        Keep the string as short as possible and use kebab case, starting with a lowercase letter.
+
+        Example: `"temp-vertex-hpt-test"`
+        """
+        pass
+
+    def setup_method(self):
+        importlib.reload(initializer)
+        importlib.reload(aiplatform)
+
+    @pytest.fixture()
+    def shared_state(self):
```

Comment:
I think type hints would be useful for the base class so eng further understands the usage of the base class. 

```suggestion
    def shared_state(self) -> Generator[Dict[str, Any], None, None]:
```